src/JSInterop/Microsoft.JSInterop/ref/Microsoft.JSInterop.netcoreapp3.0.cs

```diff
@@ -50,7 +50,7 @@ public abstract partial class JSRuntime : Microsoft.JSInterop.IJSRuntime
         protected System.TimeSpan? DefaultAsyncTimeout { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } [System.Runtime.CompilerServices.CompilerGeneratedAttribute]set { } }
         protected internal System.Text.Json.JsonSerializerOptions JsonSerializerOptions { [System.Runtime.CompilerServices.CompilerGeneratedAttribute]get { throw null; } }
         protected abstract void BeginInvokeJS(long taskId, string identifier, string argsJson);
-        protected internal abstract void EndInvokeDotNet(string callId, bool success, object resultOrError, string assemblyName, string methodIdentifier, long dotNetObjectId);
+        protected internal abstract void EndInvokeDotNet(Microsoft.JSInterop.Infrastructure.JSCallInfo callInfo, in Microsoft.JSInterop.Infrastructure.DotNetInvocationResult dispatchResult);
```

Comment:
The design in the API review had things like error info in the `CallInfo` item. But that felt like conflating inputs and outputs. This turned out to be slightly more reasonable design. We can produce a `JSCallInfo` at the onset and pass it thru the stack. It has everything we need (so far) to correctly identify a call, wheras the latter is the success \ failure discriminated union.

I haven't, as yet, done the work to correctly populate the "Error Kind". As of now, we catch the result of the entire invocation - it's a fair bit of work to get fine grained error kinds that identify if it's a parse versus invocation versus bad object id. I wanted to get the API design in first, we can always do the identification bit in 3.1 since it doesn't affect the shape of this API.